src/reducers/appReducer.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2015 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+// TODO: Remove next line
+/* eslint-disable @typescript-eslint/no-explicit-any -- included for conservative refactoring to typescript */
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import type { RootState } from '.';
+
+interface AppState {
+    portName: null | any;
+    metadata: any[];
+    isSmuMode: boolean;
+    deviceRunning: boolean;
+    rttRunning: boolean;
+    advancedMode: boolean;
+    capabilities: any;
+    samplingRunning: boolean;
+    isSaveChoiceDialogVisible: boolean;
+    isExportDialogVisible: boolean;
+    fileLoaded: boolean;
+}
+
+const initialState = (): AppState => ({
+    portName: null,
+    metadata: [],
+    isSmuMode: false,
+    deviceRunning: false,
+    rttRunning: false,
+    advancedMode: false,
+    capabilities: {},
+    samplingRunning: false,
+    isSaveChoiceDialogVisible: false,
+    isExportDialogVisible: false,
+    fileLoaded: false,
+});
+
+const appSlice = createSlice({
+    name: 'app',
+    initialState: initialState(),
+    reducers: {
+        deviceOpenedAction: (
+            state,
+            action: PayloadAction<{
+                capabilities: any;
+                portName: any;
+            }>
+        ) => {
+            state.portName = action.payload.portName;
+            state.capabilities = action.payload.capabilities;
+        },
+        deviceClosedAction: () => initialState(),
+        setDeviceRunningAction: (
+            state,
+            action: PayloadAction<{ isRunning: boolean }>
```

Comment:
You probably know that if an action just takes a single thing as payload, you could do without the object and say `action: PayloadAction<boolean>`. 

But having an additional object as you do is also fine and may have the benefit to express more clearly the meaning of the parameter being passed.

Reply:
Didn't know, so thanks for letting me know :)